api/src/routing.js

```diff
@@ -154,6 +153,12 @@ app.get('/', function(req, res) {
 });
 
 app.get('/favicon.ico', (req, res) => {
+  // Cache for a week.  Normally we don't interferse with couch headers, but
+  // due to Chrome (including Android WebView) aggressively requesting
+  // favicons on every page change and window.history update
+  // (https://github.com/medic/medic-webapp/issues/1913 ), we have to stage an
+  // intervention:
+  writeHeaders(req, res, [['Cache-Control', 'public, max-age=604800']]);
```

Comment:
It looks like you've got some other changes included in this PR. This makes the review more difficult...